### PR TITLE
GUI Updates [Menubar, Auto-Updater]

### PR DESCRIPTION
**Pull request by hoffmabc**
This update will include:
- Updated menu bar and tray icon
- Auto-updater

### CODE REVIEW

**Comment by hoffmabc** [comment]
Hey @jjeffryes this is not perfect yet, but it's getting insane to keep rebasing and keeping up to date. Can we test and merge for now?

**Comment by hoffmabc** [comment]
I'll try to squash the weird commits

**Comment by hoffmabc** [comment]
Ok @jjeffryes ready to go

File: main.js [hunk 1]
```diff
@@ -84,6 +108,143 @@ app.on('before-quit', function (e) {
...
+    ]
+  }
+  ]);
+  menu.setApplicationMenu(appMenu);
+
+  trayMenu = new tray(__dirname + '/imgs/menubar_icon.png');
+  var contextMenu = menu.buildFromTemplate([
+    {
+      label: 'Start Local Server', type: 'normal', click: function () {
+      start_local_server();
```
Comment (jjeffryes) [inline main.js:219]: The command to start the server doesn't appear to work in Windows 10.

File: main.js [hunk 2]
```diff
@@ -84,6 +108,143 @@ app.on('before-quit', function (e) {
...
+        } else {
+          console.log('Server does not seem to be running.');
+        }
+      });
+    }
+    },
+    {type: 'separator'},
+    {label: 'View Debug Log', type: 'normal', click: function() {
+      // Open Debug Log Wherever It Is
+
```
Comment (jjeffryes) [inline main.js:236]: I suppose this line needs to be more than a comment to make opening the debug logs work. ;)

Comment (hoffmabc) [inline main.js:236]: Yes I should have labeled this 1 of 2.

**Pull request merged**